internal/bminventory/inventory.go

```diff
@@ -73,6 +73,7 @@ var (
 	DefaultClusterNetworkCidr       = "10.128.0.0/14"
 	DefaultClusterNetworkHostPrefix = int64(23)
 	DefaultServiceNetworkCidr       = "172.30.0.0/16"
+	DefaultNTPSource                = "clock.redhat.com"
```

Comment:
How will it work with on prem? i don't think that we have access to `redhat.com`

Reply:
@avishayt @mhrivnak how do you think it should be handled?

Reply:
There are two questions here
1. Can we push the PR as it is right now and handle on-prem limitations later
2. How to handle onprem for the longer term.

Reply:
IMO before setting it as the default we should call validations.ValidateNTPSource(), and if it fails have no NTP source for the cluster.  This means that the validations can still pass if the hosts don't get an NTP source from their environment, right?

Reply:
I guess this is a better default than redhat.com:  https://www.ntppool.org/en/use.html

Reply:
In terms of picking a default, that decision has already been made for RHEL, taking into consideration supportability and customer requirements. We should just re-use what they decided. Their defaults are:

```
0.rhel.pool.ntp.org
1.rhel.pool.ntp.org
2.rhel.pool.ntp.org
3.rhel.pool.ntp.org
```

Reply:
@avishayt looks like this PR is doing a validation for none default NTP, but what will happen if the default NTP service is down, do we want to fail cluster creation?

Reply:
> There are two questions here
> 
> 1. Can we push the PR as it is right now and handle on-prem limitations later

No. We at least need a default that's available on the public internet.

Customers deploying bare metal will always have their hosts "on-prem". What limitations are you referring to exactly? Are you asking about disconnected/restricted networks?

> 2. How to handle onprem for the longer term.

 

I like Avishay's idea to check if an ntp server is already configured and working.

Reply:
Why do we need to validate that there is a default?  That default might not even be reachable (e.g., firewalled).  We should validate that each host has an NTP source, whether it is set via API or via DHCP.

Reply:
Yes i'm talking about disconnected, but all we need to validate that it have some NTP configured then it's fine, the validation will just fail because NTP is not reachable.

Reply:
> IIUC this is just a default, and the user can provide their own, right? So if they can't reach the default, validation will fail, and they can provide a different value?
> 
> As a default, it would be better to take an NTP value from DHCP if available. But that could be an enhancement.
> 
> What's the default NTP server in RHEL? Looks like this one is not public.
> 
> ```
> $ ntpdate -q clock.redhat.com
> Error resolving clock.redhat.com: Name or service not known (-2)
> 24 Nov 15:26:19 ntpdate[10497]: Can't find host clock.redhat.com: Name or service not known (-2)
> 24 Nov 15:26:19 ntpdate[10497]: no servers can be used, exiting
> ```

@mhrivnak 
`clock.redhat.com` is an NTP pool. I took it from [ocs-ci repository](https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/constants.py#L1032)

> @avishayt looks like this PR is doing a validation for none default NTP, but what will happen if the default NTP service is down, do we want to fail cluster creation?

@filanov 
In case of the default failure, I don't think we should fail. It might be because of firewall rules.
The hosts would be insufficient and the discovery stage would hang until a new NTP source would be added.

> Why do we need to validate that there is a default? That default might not even be reachable (e.g., firewalled). We should validate that each host has an NTP source, whether it is set via API or via DHCP.

@avishayt 
This is exactly what is happening right now.
The cluster `ntpSource` field is just added to the agent pool of ntp sources to sync with. Each host needs to be synced with **any** server.

Reply:
This is my understanding of the behavior - if it's correct then I think this PR is fine:
1. There is a default NTP source, which may or not be reachable.  There is no validation on that.
2. We detect the NTP source for each host, and there is a validation on that.  The host can get NTP either from DHCP or the default source.
3. The user can change the default source, which will cause the hosts to use that, and update the validation.

@YuviGold Correct?

Reply:
> This is my understanding of the behavior - if it's correct then I think this PR is fine:
> 
> 1. There is a default NTP source, which may or not be reachable.  There is no validation on that.
> 2. We detect the NTP source for each host, and there is a validation on that.  The host can get NTP either from DHCP or the default source.
> 3. The user can change the default source, which will cause the hosts to use that, and update the validation.
> 
> @YuviGold Correct?

That's right.